File: architect_zenghui/src/main/java/com/east/architect_zenghui/architect12_Designmode5_tempolate/BaseActivity.java
package com.east.architect_zenghui.architect12_Designmode5_tempolate;

import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

/**
 * |---------------------------------------------------------------------------------------------------------------|
 *
 *  @description: Activity的基类,使用模板设计模式
 *  @author: East
 *  @date: 2020-02-19
 * |---------------------------------------------------------------------------------------------------------------|
 */
public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // 1.设置布局
        setContentView();
        // 写一些公用的方法，ButterKnife注解，统一管理Activity，等等
        // 2. 初始化Title
        initTitle();
        // 3. 初始化View
        initView();
        // 4. 访问接口数据（initData）
        initData(savedInstanceState);
    }

    protected abstract void setContentView();

    protected abstract void initTitle();

    protected abstract void initView();

    protected abstract void initData(Bundle savedInstanceState);

    //写一大堆方法 ，很多 Activity 都要用的
    public void startActivity(Class<? extends BaseActivity> clazz){
        startActivity(new Intent(this,clazz));
    }

}

File: 2.曾辉笔记/zh-1.内涵段子/code/ConnotationJokes/baselibrary/src/main/java/com/east/baselibrary/base/BaseActivity.kt
package com.east.baselibrary.base

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.east.baselibrary.ioc.JaminIoc

/**
 * |---------------------------------------------------------------------------------------------------------------|
 *  @description: Activity的基类
 *  @author: jamin
 *  @date: 2020/4/30
 * |---------------------------------------------------------------------------------------------------------------|
 */
abstract class BaseActivity: AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(getLayoutId())
        //加入自己的IOC注解
        JaminIoc.bind(this)
        initTitle()//初始化标题
        initView()//初始化界面
        initData()//初始化数据
    }

    //获取布局文件的ID
    protected abstract fun getLayoutId(): Int

    //初始化标题
    protected abstract fun initTitle()

    //初始化界面
    protected open fun initView(){}

    //初始化数据
    protected abstract fun initData()

    protected fun startActivity(clazz:Class<*>){
        startActivity(Intent(this,clazz))
    }

}